low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -0,0 +1,467 @@
+        """Bind the service to a signal('PropertiesChanged') which will be
+           triggered whenever the service changes it's state/substate.
+        """
+        try:
+            unit, _, state, substate, pid = self.get_service_status(service=service)
+
+            self.update_status(service, state, substate, pid)
+
+            Iunit2 = Interface(unit,
+                            dbus_interface='org.freedesktop.systemd1.Manager')
+
+            Iunit2.connect_to_signal('PropertiesChanged',
+                                    lambda a, b, c, p = unit :
+                                    self.on_prop_changed(a, b, c, p),
+                                    dbus_interface=PROPERTIES_IFACE)
+
+            logger.debug(f"{service}({pid}) state is {state}:{substate}")
+
+            if state in  ["activating", "reloading", "deactivating"]:
+                self.not_active_services[service] = \
+                                    [self.current_time(), "N/A", "N/A"]
+            elif state != "active":
+                self.failed_services.append(service)
+                self.raise_alert(service, "N/A", state, "N/A", substate,
+                                    "N/A", pid, 0)
+                logger.error(f"{service} is not active initially. state = {state}:{substate}")
+
+            return None
+        except DBusException as err:
+            return err
+
+    def check_notactive_services(self):
+        """Raise Alert if any of the not-active services has exceeded
+            the threshould time for inactivity.
+        """
+        not_active_services_copy = self.not_active_services.copy()
+        for service, [start_time, prev_state, prev_substate]\
+                                 in not_active_services_copy.items():
+
+            if self.current_time() - start_time > self.max_wait_time:
+                state = self.service_status[service]["state"]
+                substate = self.service_status[service]["substate"]
+                pid = self.service_status[service]["pid"]
+                self.not_active_services.pop(service)
+                self.failed_services.append(service)
+                self.raise_alert(service, prev_state, state, prev_substate,
+                                 substate, pid , pid, 1)
+
+    def update_status(self, service, state, substate, pid):
+        self.service_status[service] = {
+            "state" : state,
+            "substate" : substate,
+            "pid" : pid
+        }
+
+    def on_prop_changed(self, interface, changed_properties,
+                                invalidated_properties, unit):
+        """Handler to process the service state change signal."""
+
+        _, service, state, substate, pid = self.get_service_status(unit=unit)
+
+        prev_state = self.service_status[service]["state"]
+
+        logger.debug(f"Event for {service}, properties changed from "\
+                     f"{prev_state} to {state}")
+
+        if prev_state == state:
```

Comment:
mostly sub-state changed? log it then with logger.debug

Reply:
sorry didn't get it.  Do you want debug-log at line 271 or something else logged?

Reply:
yeah am saying we can log atleast, with logger.debug for any sub-state change !?

Reply:
done.. added substate to debug log.